### DISCUSSION TITLE
How to Revalidate a Specific Dynamic Segment (/product/123) in Next.js Without Revalidating All Segments (/product/[id])? - Next.js 13.4.6

### DISCUSSION THREAD

**Discussion by sadeghnakhjavani** [vercel/next.js#52385] [App Router]
I am working on a Next.js project version 13.4.6 and I have a dynamic route /product/[id]. I need to revalidate the data for a specific product, for example, /product/123, when its content changes.

I am aware of the revalidatePath function, which allows revalidating data associated with a specific path. However, when I use revalidatePath with a dynamic segment such as revalidatePath('/product/[id]'), it revalidates all segments under /product/[id].

The Next.js version 13.4.6 documentation says that the revalidatePath function accepts a single parameter, path, which should be a string representing the filesystem path associated with the data you want to revalidate. This is not the literal route segment(e.g. /product/123), but instead, the path on the filesystem (e.g. /product/[id])​​.

What I want to achieve is to selectively revalidate only a specific product ID, like /product/123, without revalidating all the other segments under /product/[id].

Is there a way to achieve this selective revalidation in Next.js 13?

**Comment by icyJoseph** [comment]
If you use `fetch`, is it possible for you to use `tags`? And `revalidateTag` instead? https://nextjs.org/docs/app/api-reference/functions/revalidateTag

**Reply by sadeghnakhjavani** [reply]
Thank you for you response...
Assume that I have 1000 products.
As you said I should specify 1 unique tag - created by product id - for each product when I fetch the content of the product.
To revalidate product content I should revalidate that unique tag.
Is this true ?

**Reply by icyJoseph** [reply]
Yeah, at least that's how I interpret tags, they allow you to have a more granular revalidation strategy. The tag could just be the `product/[productId]` result, as it uniquely identifies the page, it should uniquely tag the product too.

**Comment by alsherif-khalaf** [comment]
> If you use `fetch`, is it possible for you to use `tags`? And `revalidateTag` instead? https://nextjs.org/docs/app/api-reference/functions/revalidateTag

waht about using something like Prisma to get products , how can I tag it ?
for example :

```ts
  const todos = await prisma.todo.findMany();
```
